dbms/src/Server/CertificateReloader.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Server/CertificateReloader.h>
+
+#if Poco_NetSSL_FOUND
+
+#include <Poco/Net/Utility.h>
+
+namespace DB
+{
+namespace
+{
+/// Call set process for certificate.
+int callSetCertificate(SSL * ssl, void * arg)
+{
+    return CertificateReloader::instance().setCertificate(ssl, arg);
+}
+} // namespace
+
+/// This is callback for OpenSSL. It will be called on every connection to obtain a certificate and private key.
+int CertificateReloader::setCertificate(SSL * ssl, void * arg)
+{
+    auto * context = static_cast<Context *>(arg);
+    auto security_config = context->getSecurityConfig();
+    auto [cert_path, key_path] = security_config->getCertAndKeyPath();
+    LOG_INFO(log, "setCertificate callback called, cert_path: {}, key_path: {}", cert_path, key_path);
```

Comment:
use `LOG_DEBUG`?
Could it be printing too many logs?

Reply:
ssl connection build is rare. But I think LOG_DEBUG is good